src/pl/senla/hotel/application/di/DIApplication.java

```diff
@@ -0,0 +1,104 @@
+package pl.senla.hotel.application.di;
+
+import pl.senla.hotel.entity.Guest;
+import pl.senla.hotel.entity.Order;
+import pl.senla.hotel.entity.Hotel;
+import pl.senla.hotel.entity.facilities.HotelFacility;
+import pl.senla.hotel.entity.services.HotelService;
+import pl.senla.hotel.ie.serialization.Processor;
+import pl.senla.hotel.ie.serialization.ProcessorSerializable;
+import pl.senla.hotel.storage.*;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.List;
+
+public class DIApplication {
+
+    private static DIApplication application;
+    private DIApplication() {}
+    public static synchronized DIApplication getApplication() {
+        if (application == null) {
+            application = new DIApplication();
+        }
+        return application;
+    }
+
+    public static void run() throws InvocationTargetException,
+            IllegalAccessException, InstantiationException, NoSuchMethodException {
+        DIContext context = DIContext.getContext();
+        context.createDIContainer();
+        context.injectValuesFromDIContainer();
+        loadHotelData(context);
+        runApplication(context);
+        saveHotelData(context);
+        quitHotel();
```

Comment:
why do you quit here? as i remember it should be an infinity loop and you should manage it from one of the menu options.

Reply:
I changed logic of MainMenu where was all logic for quit from application and save hotel's data before quit.  

Now my application works while method **runApplication(context)**  is running. This method invokes an endless loop of MainMenu. When the user choose "EXIT" the loop will be terminated, method **runApplication(context)** will be executed and methods  **saveHotelData(context),  quitHotel()** will save data and close application.

I'll return old logic if previous logic was correctly.